### PR TITLE
Model origin and FOV aimbot calculation

### PR DESCRIPTION
**Pull request by Doctor-Coomer**
Hello, this is one of the feature improvement PRs I was referring too :D I had found some more than edge case issues with the player origin offset, and how aimbot selection was being calculated. 

In this patch I've changed the `get_ent_origin` method to return a player origin that is more accurate. The original offset being used was problematic, as it was intended by the game for world base calculations, and the location was projected ahead of where the player actually is. This caused the ESP, and aimbot to an extent as far as I can tell from usage, to be inaccurate. The offset I've applied in this patch returns the correct entity origin.

Another patch I have added is to the aimbot, and how it selects which player to target. The original method of calculating the target had some mathematical edge cases. I have changed it to use something closer to a Field Of View calculation that uses both pitch and yaw.

### CODE REVIEW

File: source_sdk/entity/entity.c [hunk 1]
```diff
@@ -32,10 +32,10 @@ __int32_t get_ent_class(void *entity)
 
 struct vec3_t get_ent_origin(void *entity)
 {
-    // x + 0x340, y + 0x344, z + 0x348
-    return *(struct vec3_t *)((__uint64_t)(entity) + 0x340);
-}
-
+    // x + 0x328, y + 0x332, z + 0x346
+    return *(struct vec3_t *)((__uint64_t)(entity) + 0x328); //PR comment: This is the correct offset. The one which was used prior is used for world origin.
```
Comment (faluthe) [inline source_sdk/entity/entity.c]: Nice find. This works great for hitscan weapons and better for projectile aimbot too. At the moment the projectile aimbot misses at large distances, I think eventually the goal is to use `EstimateAbsVelocity` for the projectile prediction.

**Comment by Doctor-Coomer** [comment]
I've messed up. I pushed my name esp to the wrong branch! Sorry, I'm bad at git, I'll try to fix it.

**Comment by faluthe** [comment]
> I've messed up. I pushed my name esp to the wrong branch! Sorry, I'm bad at git, I'll try to fix it.

I can fix it now.

**Comment by faluthe** [comment]
@yoshisaac Fantastic changes! Tested and working good on my system. Nice find with the new offsets. I've removed the PR comments and I think its ready for a merge.

**Comment by Doctor-Coomer** [comment]
> I can fix it now.

Thank you so much :pray:

**Comment by Doctor-Coomer** [comment]
> @yoshisaac Fantastic changes! Tested and working good on my system. Nice find with the new offsets. I've removed the PR comments and I think its ready for a merge.

Thank you. Sounds great to me.

**Pull request merged**